libsolidity/formal/SMTEncoder.cpp

```diff
@@ -861,6 +857,9 @@ void SMTEncoder::endVisit(Identifier const& _identifier)
 		defineExpr(_identifier, m_context.state().thisAddress());
 		m_uninterpretedTerms.insert(&_identifier);
 	}
+	// Ignore struct type identifiers in struct constructor calls
+	else if (auto typetype = dynamic_cast<TypeType const*>(_identifier.annotation().type); typetype && typetype->actualType()->category() == Type::Category::Struct)
+		return;
```

Comment:
@leonardoalt I am not sure about this one, but otherwise it would fall through to the call to createExpression, which is not what we want. The encoder should not try to create an expression for `S` in `S(1,2)`.

Reply:
I think this should be fine. Also, if you remove the last part of the condition, it could be that it also solves https://github.com/ethereum/solidity/issues/10141: the issue there is that the warning message is spurious, the conversion already works

Reply:
OK, I can try that.

Reply:
When we ignore all TypeType identifiers in this way, there is quite a lot of tests where it removes warnings like `Assertion checker does not yet implement type type(library L)`. Beside library it happens also for `enum` and contract types. Apart from that, the tests still seem to go through fine. So I am wondering why this has not been done before and if I am not missing something. Since it is an unrelated change, maybe we should it in a separate PR?

Reply:
This was on the list to be done, I don't see why it shouldn't. Agree that it should go in a different PR